File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/Commands/UpdateOrderStatusCommand.cs
using MediatR;
using MediatorDesignPattern.Example2.Models;

namespace MediatorDesignPattern.Example2.Commands;

/// <summary>
/// Command to update order status
/// </summary>
public class UpdateOrderStatusCommand : IRequest<Order?>
{
    public Guid OrderId { get; set; }
    public OrderStatus NewStatus { get; set; }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/src/Commands/UpdateOrderStatusCommand.ts
import { RequestData } from 'mediatr-ts';
import { Order, OrderStatus } from '../Models/Order';

export class UpdateOrderStatusCommand extends RequestData<Order | undefined> {
  orderId: string;
  newStatus: OrderStatus;

  constructor(orderId: string, newStatus: OrderStatus) {
    super();
    this.orderId = orderId;
    this.newStatus = newStatus;
  }
}
